src/basic/fs-util.c

```diff
@@ -713,6 +713,9 @@ int chase_symlinks(const char *path, const char *original_root, unsigned flags,
                                  * what we got so far. But don't allow this if the remaining path contains "../ or "./"
                                  * or something else weird. */
 
+                                if (streq_ptr(done, "/"))
+                                        *done = '\0';
```

Comment:
can you elaborate what this does?

Reply:
Without this, `chase_symlinks("/../../foo/bar", ...)` returns `//foo/bar`. This removes the unnecessary slash at the head.

Reply:
I've commented this in the revised commit.

Reply:
Also commented in the source.